### PR TITLE
Add cita formula.

### PR DESCRIPTION
**Pull request by clearloop**
## CircleCi
well, the circleCI seems failed with connecting nodes in some tests...XD, I'll try to fix them the day after tomorrow If they're not network errors. (I have to write my graduation thesis tomorrow)

## Diff
1. Added `cita.rb`
2. Modified the way of getting absolute path from `$(cd $(dirname $0) && pwd -P )` into `realpath` and `readlink -f`, because the `pwd` method still not friendly to symlinks.
3. Modified `cita-forever`'s stdout to `cita-forever -c ${config} start > /dev/null 2>&1` in `cita script`, I'm not sure about this, but `2>&1` directly causes a bug in my PC —— keyboard stdin messed up after `cita start`, `cita ping` will not work...XD(might be my last pr's fault)

## Note
1. This script choose `secp256k1` algorithm by default, the reason is the `secp256k1` versioned `cita-cli` installing easily, and `cita-sdk-js` might not be happy with `ed25519` [recently][1]...
2. May be cita team can new a repo named `homebrew-cita` so users can use `brew tap cryptape/cita` installing cita directly, or pr the `cita.rb` to [homebrew-core][2] after cita have released stable version? `brew install cita` directly might be the best choice~

#### Install docker
_To run cita happily, we need to install docker first._
```
brew cask install docker
```

#### Curl formula and install.

```sh
# curl https://raw.githubusercontent.com/clearloop/cita/develop/formula/cita.rb -o cita.rb
curl https://raw.githubusercontent.com/cryptape/cita/develop/formula/cita.rb -o cita.rb
brew install ./cita.rb
```

By default, binaries installed by cita will be placed into:
  
```
/usr/local/Cellar/cita/0.22
```

Run `cita help` for more detailed information.

#### One more thing

Click docker icon, and share `/usr/local/Cellar/cita`.

#### Create Chain

```
$ cita create --chain_name "cita_block_chain" --super_admin "0x4b5ae4567ad5d9fb92bc9afd6a657e6fa13a2523" --nodes "127.0.0.1:4000,127.0.0.1:4001,127.0.0.1:4002,127.0.0.1:4003"
```

#### Setup Chain

```
$ cita setup test-chain/0
```

#### Start Chain
```
$ cita start test-chain/0
```

#### Stop Chain
```
$ cita stop test-chain/0
```

#### Usages
```
$ cita
Usage: cita <command> <node> [options]
where <command> is one of the following:
    { help | create | port | setup | start | stop | restart
      ping | top | backup | clean | logs | logrotate }
Run `cita help` for more detailed information.
```

## Presets

_Here are some default sets, you can reset them on your own:_

#### Set Port

```
$ cita port 42:42
# The docker port is `1337:1337` by default, if you need to, use this._
```
[1]: https://talk.citahub.com/t/topic/334
[2]: https://github.com/Homebrew/homebrew-core

### CODE REVIEW

**Comment by kaikai1024** [comment]
> Modified cita-forever's stdout to cita-forever -c ${config} start > /dev/null 2>&1 in cita script, I'm not sure about this, but 2>&1 directly causes a bug in my PC

check the #367 , the log can out to the stdout, so remove the `> dev/null`.

> keyboard stdin messed up after cita start, cita ping will not work

#385 This?

> May be cita team can new a repo named homebrew-cita so users can use brew tap cryptape/cita installing cita directly, or pr the cita.rb to homebrew-core after cita have released stable version? 

I am not familar with the homebrew, I'll do a research. thanks.

btw,  congratulations about the coming graduation!

File: formula/cita.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# CITA Formula
```
Comment (leeyr338) [inline formula/cita.rb]: @kaikai1024 , I think is better to let this installation-related files in a separate repository. what do you think?

Comment (kaikai1024) [reply formula/cita.rb]: yeah. seems that's better.
i will check it later

**Comment by kaikai1024** [comment]
> 2>&1 directly causes a bug in my PC —— keyboard stdin messed up after cita start, cita ping will not work

btw,  you should not change the code, you need to open an new issue about this.  #384

**Comment by kaikai1024** [comment]
> May be cita team can new a repo named homebrew-cita

We accept that. Thanks :heart: , you can pull request on that repo, close it and if you want to update the shell script, create a new pr please.

**Pull request closed without merging**